docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -156,262 +208,994 @@ bootstrapTokens:
+          defaultMode: 256
+  volumeClaimTemplates:
+  - metadata:
+      name: mongodb-persistent-storage-claim
+      annotations:
+        volume.beta.kubernetes.io/storage-class: "mongodb-sc"
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+```
+$ kubectl create -f mongodb-statefulset.yaml
+statefulset.apps/mongo created
+```
+Verify that the MongoDB application has been deployed.
+Wait for pods to start running and PVCs to be created for each replica. 
+```
+$ kubectl get statefulset mongod
+NAME     READY   AGE
+mongod   3/3     96s
+```
+```
+$ kubectl get pod -l role=mongo
+NAME       READY   STATUS    RESTARTS   AGE
+mongod-0   1/1     Running   0          13h
+mongod-1   1/1     Running   0          13h
+mongod-2   1/1     Running   0          13h
+```
+```              
+$ kubectl get pvc
+NAME                                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+mongodb-persistent-storage-claim-mongod-0   Bound    pvc-ea98b22a-b8cf-11e9-b1d3-005056a0e4f0   1Gi        RWO            mongodb-sc     13h
+mongodb-persistent-storage-claim-mongod-1   Bound    pvc-0267fa7d-b8d0-11e9-b1d3-005056a0e4f0   1Gi        RWO            mongodb-sc     13h
+mongodb-persistent-storage-claim-mongod-2   Bound    pvc-24d86a37-b8d0-11e9-b1d3-005056a0e4f0   1Gi        RWO            mongodb-sc     13h
+```
+### Set up the Mongo replica set configuration
+To setup the Mongo replica set configuration, we need to connect to one of the mongod container processes to configure the replica set.
+Run the following command to connect to the first container. In the shell, initiate the replica set. You can rely on the host names to be the same, due to having employed the StatefulSet.
+```
+$ kubectl exec -it mongod-0 -c mongod-container bash
+root@mongod-0:/# mongo
+> rs.initiate({_id: "MainRepSet", version: 1, members: [
+... { _id: 0, host : "mongod-0.mongodb-service.default.svc.cluster.local:27017" },
+... { _id: 1, host : "mongod-1.mongodb-service.default.svc.cluster.local:27017" },
+... { _id: 2, host : "mongod-2.mongodb-service.default.svc.cluster.local:27017" }
+...  ]});
+{ "ok" : 1 }
 ```
+This makes mongodb-0 the primary node and other two nodes are secondary.
+### Verify Cloud Native Storage functionality is working in vSphere
+After your application gets deployed, its state is backed by the VMDK file associated with the specified storage policy. As a vSphere administrator, you can review the VMDK that is created for your container volume.
+In this step, we will verify that the Cloud Native Storage feature released with vSphere 6.7U3 is working. To go to the CNS UI, login to the vSphere client, then navigate to Datacenter → Monitor → Cloud Native Storage → Container Volumes and observe that the newly created volume are present along with all labels. You can also monitor their storage policy compliance status.
```

Comment:
Get a screenshot in here - looks awesome.

Reply:
Done.